src/metrics/src/rusage.rs

```diff
@@ -5,55 +5,99 @@
+                    let ret = libc::getrusage(libc::RUSAGE_SELF, &mut rusage);
+                    if ret < 0 {
+                        return;
+                    }
                     rusage
                 };
-                $(self.$name.set(rusage.$name);)*
+                $(self.$name.set(<$type as Unit>::from(rusage.$name));)*
             }
         }
     };
 }
 
+/// Type for converting values from POSIX to Prometheus.
+trait Unit {
+    /// Prometheus gauge
+    type Gauge;
+    /// Libc type
+    type From;
+    /// Gauge type.
+    type To;
+    /// Convert an actual value.
+    fn from(value: Self::From) -> Self::To;
+}
+
+/// Unit for converting POSIX timeval.
+struct Timeval;
+impl Unit for Timeval {
+    type Gauge = Gauge;
+    type From = libc::timeval;
+    type To = f64;
+    fn from(Self::From { tv_sec, tv_usec }: Self::From) -> Self::To {
+        // timeval can capture negative values; it'd be surprising to see a negative values here.
+        Duration::from_secs(tv_sec.abs_diff(0))
+            .add(Duration::from_micros(tv_usec.abs_diff(0)))
+            .as_secs_f64()
+    }
+}
+
+/// Unit for direct conversion to i64.
+struct Unitless;
+impl Unit for Unitless {
+    type Gauge = IntGauge;
+    type From = libc::c_long;
+    type To = i64;
+    fn from(value: Self::From) -> Self::To {
+        value
+    }
+}
+
 metrics! {
     mz_metrics_libc
-    (ru_maxrss, "maximum resident set size"),
-    (ru_ixrss, "integral shared memory size"),
-    (ru_idrss, "integral unshared data size"),
-    (ru_isrss, "integral unshared stack size"),
-    (ru_minflt, "page reclaims (soft page faults)"),
-    (ru_majflt, "page faults (hard page faults)"),
-    (ru_nswap, "swaps"),
-    (ru_inblock, "block input operations"),
-    (ru_oublock, "block output operations"),
-    (ru_msgsnd, "IPC messages sent"),
-    (ru_msgrcv, "IPC messages received"),
-    (ru_nsignals, "signals received"),
-    (ru_nvcsw, "voluntary context switches"),
-    (ru_nivcsw, "involuntary context switches")
+    (ru_utime, "user CPU time used", "_s", Timeval),
+    (ru_stime, "system CPU time used", "_s", Timeval),
+    (ru_maxrss, "maximum resident set size", "", Unitless),
+    (ru_ixrss, "integral shared memory size", "", Unitless),
+    (ru_idrss, "integral unshared data size", "", Unitless),
+    (ru_isrss, "integral unshared stack size", "", Unitless),
+    (ru_minflt, "page reclaims (soft page faults)", "", Unitless),
+    (ru_majflt, "page faults (hard page faults)", "", Unitless),
+    (ru_nswap, "swaps", "", Unitless),
+    (ru_inblock, "block input operations", "", Unitless),
+    (ru_oublock, "block output operations", "", Unitless),
+    (ru_msgsnd, "IPC messages sent", "", Unitless),
+    (ru_msgrcv, "IPC messages received", "", Unitless),
+    (ru_nsignals, "signals received", "", Unitless),
+    (ru_nvcsw, "voluntary context switches", "", Unitless),
+    (ru_nivcsw, "involuntary context switches", "", Unitless)
```

Comment:
Some of these are not really unitless but, e.g., kilobytes. And some should probably also have a `_total` suffix. Could you add a TODO so we don't forget to adjust these too?

Reply:
Fixed, thanks!